code/modules/mob/living/basic/guardian/guardian_types/ranged.dm

```diff
@@ -62,6 +64,12 @@
 	sync_lighting_plane_cutoff()
 	to_chat(src, span_notice(msg))
 
+/// Safety to prevent us from using ranged attacks while unleashed. Typically the status effect will catch stuff like this but we want to be extra sure with unleashing schenanigans
+/mob/living/basic/guardian/ranged/proc/on_ranged_attack(datum/source, atom/target, modifiers)
```

Comment:
i think this is a good hook regardless in case you want to prevent people firing at the summoner or something (which is something I wrote in when I tried to do the guardian rework about 14 months ago)

Reply:
maybe another day